arch/ARM/STM32/drivers/i2c_stm32f4/stm32-i2c.adb

```diff
@@ -561,6 +561,14 @@ package body STM32.I2C is
 
             This.Periph.DR.DR := UInt8 (Mem_Addr and 16#FF#);
       end case;
+
+      --  Wait until TXE flag is set
+      Wait_Flag (This, Tx_Data_Register_Empty, False, Timeout, Status);
+
+      if Status /= HAL.I2C.Ok then
```

Comment:
Hello Hedley,

Sorry for the delay. 

This if then return is not needed. Can you modify this patch or add the new commit that remove those 3 lines?

Thanks in advance,